File: src/main/java/EventExample.java
import java.io.IOException;

/**
 * This example shows how to use the event feature.
 * The database server must be started first to make this example work.
 * Documentation: http://basex.org/api
 *
 * @author BaseX Team 2005-11, BSD License
 */
public final class EventExample {

  /**
   * Main method.
   * @param args command-line arguments
   */
  public static void main(final String[] args) {
    new EventExample();
  }

  /** Constructor. */
  private EventExample() {
    try {
      BaseXClient session1 =
        new BaseXClient("localhost", 1984, "admin", "admin");
      BaseXClient session2 =
        new BaseXClient("localhost", 1984, "admin", "admin");

      session1.execute("create event messenger");
      session2.watch("messenger", new Notifier());
      session2.query("for $i in 1 to 1000000 where $i = 0 return $i").execute();
      session1.query("db:event('messenger', 'Hello World!')").execute();
      session2.unwatch("messenger");
      session1.execute("drop event messenger");
      session1.close();
      session2.close();

    } catch(IOException e) {
      e.printStackTrace();
    }
  }

  /**
   * Implementation of the event notifier interface.
   */
  private class Notifier implements BaseXClient.EventNotifier {

    /** Constructor. */
    public Notifier() { }

    @Override
    public void notify(final String value) {
      System.out.println("Message received: " + value);
    }
  }
}

File: src/main/c#/EventExample.cs
/*
 * This example shows how to use the event feature.
 *
 * Documentation: http://docs.basex.org/wiki/Clients
 *
 * (C) BaseX Team 2005-12, BSD License
 */
using System;
using System.Diagnostics;
using System.IO;

namespace BaseXClient
{
  public class EventExample
  {
    public static void Main(string[] args)
    {
      try
      {
        // create session
        Session session1 = new Session("localhost", 1984, "admin", "admin");
        Session session2 = new Session("localhost", 1984, "admin", "admin");

        session1.Execute("create event messenger");
        session2.Watch("messenger", new Notification());
        session2.Query("for $i in 1 to 1000000 where $i = 0 return $i").Execute();
        session1.Query("db:event('messenger', 'Hello World!')").Execute();
        session2.Unwatch("messenger");
        session1.Execute("drop event messenger");

        // close session
        session1.Close();
        session2.Close();
      }
      catch (IOException e)
      {
        // print exception
        Console.WriteLine(e.Message);
      }
    }
  }
  
  class Notification : EventNotification
  {
    public void Update (string data)
    {
      Console.WriteLine("Message received: " + data);
    }
  }
}